### DISCUSSION TITLE
Can we add similar solutions together?

### DISCUSSION THREAD

**Discussion by cyasing** [pybamm-team/PyBaMM#2977] [Q&A]
I am running a simulation where the current input which depends on time in simulation. Basically, for each second, a different current input is given like this: `solution = sim.solve([t,t+1],inputs={"Current function [A]" : I]})` where I changes after each second of simulation. Due to it not being an experiment, I cannot use `starting_solution` argument while solving (I do use `set_initial_conditions_from` to ensure correct solution). So, is there any way in which I can "add" solutions for same cell but different times together? I can store an array of solutions, I just need to be able to quickplot it with entire time, instead of just the last one second.

**Comment by TomTranter** [comment]
try using solver.step

**Comment by rtimms** [comment]
You can also add two solutions: `new_sol = sol_1 + sol_2`, but `step` is probably the best solution for you.
